pkg/dao/admin_tasks.go

```diff
@@ -0,0 +1,140 @@
+package dao
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	ce "github.com/content-services/content-sources-backend/pkg/errors"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"github.com/content-services/content-sources-backend/pkg/pulp_client"
+	"gorm.io/gorm"
+)
+
+type adminTaskInfoDaoImpl struct {
+	db         *gorm.DB
+	pulpClient pulp_client.PulpClient
+}
+
+func GetAdminTaskDao(db *gorm.DB, pulpClient pulp_client.PulpClient) AdminTaskDao {
+	return adminTaskInfoDaoImpl{
+		db:         db,
+		pulpClient: pulpClient,
+	}
+}
+
+func (a adminTaskInfoDaoImpl) Fetch(id string) (api.AdminTaskInfoResponse, error) {
+	taskInfo := models.TaskInfo{}
+	query := a.db.Where("id = ?", id).Joins("LEFT JOIN repository_configurations ON (tasks.repository_uuid = repository_configurations.repository_uuid AND tasks.org_id = repository_configurations.org_id)")
+	result := query.First(&taskInfo)
+	var account_id sql.NullString
+	query.Select("account_id").First(&account_id)
+
+	taskInfoResponse := api.AdminTaskInfoResponse{}
+
+	if result.Error != nil {
+		if result.Error == gorm.ErrRecordNotFound {
+			return taskInfoResponse, &ce.DaoError{NotFound: true, Message: "Could not find task with UUID " + id}
+		} else {
+			return taskInfoResponse, result.Error
+		}
+	}
+
+	parsedPayload, err := taskInfo.ParsePayload(a.pulpClient)
```

Comment:
I see the pulp task information overrides the original payload of the task when returning a response here. 

I think it would make sense to have the pulp information be its own field. It might actually be useful to see the original payload, especially in the case of a half completed task.